hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java

```diff
@@ -73,6 +75,7 @@
     private final AtomicBoolean joinedJob = new AtomicBoolean();
     private final BiConsumer<Void, Throwable> joinJobCallback;
 
+    /** Used as a cache for {@link #getName()} and may be outdated. */
     private volatile JobConfig jobConfig;
```

Comment:
if we need only the name, there is no point in keeping entire `JobConfig` (with resources etc)